ui/tests/components/table/groupings-table.test.tsx

```diff
@@ -0,0 +1,187 @@
+            expect(screen.getByTestId(`clipboard-icon-${index}`)).toBeInTheDocument();
+        });
+
+        // check for pagination
+        expect(screen.getByText('First')).toBeInTheDocument();
+        expect(screen.getByText('Previous')).toBeInTheDocument();
+        expect(screen.getByText(1)).toBeInTheDocument();
+        expect(screen.getByText('Next')).toBeInTheDocument();
+        expect(screen.getByText('Last')).toBeInTheDocument();
+    });
+
+    it('filters data correctly using global filter', () => {
+        render(<GroupingsTable groupingPaths={generateSampleData} />);
+
+        const filterInput = screen.getByPlaceholderText('Filter Groupings...');
+        fireEvent.change(filterInput, { target: { value: '1' } });
+
+        expect(screen.getByText('example-1')).toBeInTheDocument();
+        expect(screen.queryByText('example-0')).not.toBeInTheDocument();
+
+        fireEvent.change(filterInput, { target: { value: '0' } });
+        expect(screen.getByText('example-0')).toBeInTheDocument();
+        expect(screen.queryByText('example-1')).not.toBeInTheDocument();
+    });
+
+    it('sorts data when header is clicked', async () => {
+        render(<GroupingsTable groupingPaths={generateSampleData} />);
+
+        // Sort by grouping name
+        await clickAndWaitForSorting('GROUPING NAME', ['example-0', 'example-1'], true);
+
+        // Descending order
+        await clickAndWaitForSorting(
+            'GROUPING NAME',
+            [`example-${generateSampleData.length - 1}`, `example-${generateSampleData.length - 2}`],
+            false
+        );
+
+        // Sort by description
+        await clickAndWaitForSorting('DESCRIPTION', ['Test Description 0', 'Test Description 1'], true);
+
+        // Descending order
+        await clickAndWaitForSorting(
+            'DESCRIPTION',
+            [`Test Description ${generateSampleData.length - 1}`, `Test Description ${generateSampleData.length - 2}`],
+            false
+        );
+
+        // Sort by grouping path
+        await clickAndWaitForSorting('GROUPING PATH', ['example-0', 'example-1'], true);
+
+        // Descending order
+        await clickAndWaitForSorting(
+            'GROUPING PATH',
+            [`example-${generateSampleData.length - 1}`, `example-${generateSampleData.length - 2}`],
+            false
+        );
+    });
+
+    it('column settings', async () => {
+        render(<GroupingsTable groupingPaths={generateSampleData} />);
+        const button = screen.getByTestId('column-settings-button');
+        const user = userEvent.setup();
+
+        const toggleColumnVisibility = async (columnTestId: string, isVisible: boolean) => {
+            await waitFor(async () => {
+                await user.click(button);
+            });
+            fireEvent.click(screen.getByTestId(columnTestId));
+            if (isVisible) {
+                // check getByText('DESCRIPTION') or getBYText('GROUPING PATH') to be in document
```

Comment:
Capitalize "c" in "check" here. and uncapitalize the "Y" in getBYText('GROUPING PATH').

Reply:
done